File: src/CarLot.Catalog/CarLot.Catalog.Application/DTOs/AddCarRequest.cs
using CarLot.Catalog.Domain.Enums;

namespace CarLot.Catalog.Application.DTOs;

public record AddCarRequest(
    string Vin,
    string Make,
    string Model,
    int Year,
    FuelType FuelType,
    AdditionalFuelType AdditionalFuelType,
    TransmissionType Transmission,
    int PowerHp,
    float? EngineDisplacement,
    bool Turbocharged,
    string Body,
    string RegistrationPlate,
    Domain.Enums.DriveType DriveType,
    int MileageKm,
    string Location,
    IEnumerable<string> EquipmentCodes);

File: src/carlot.ui/src/types/AddCarRequest.ts
import type { AdditionalFuelType, DriveType, FuelType, TransmissionType } from "./CarDto";

export interface AddCarRequest {
  vin: string;
  make: string;
  model: string;
  year: number;
  fuelType: FuelType;
  additionalFuelType: AdditionalFuelType;
  transmission: TransmissionType;
  powerHp: number;
  engineDisplacement?: number;
  turbocharged: boolean;
  body: string;
  registrationPlate: string;
  driveType: DriveType;
  mileageKm: number;
  location: string;
  equipment: ReadonlyArray<string>;
}